### DISCUSSION TITLE
Migration from vuex-orm, vuex store and model were linked

### DISCUSSION THREAD

**Discussion by jpmc3630** [CodeDredd/pinia-orm#203] [Q&A] [answered]
Thanks for answering my questions, this migration is a huge job so it really helps to clarify a few things.

In vuex-orm we had vuex stores linked with the vuex-orm models. This happened when the models were registered, which no longer happens in pinia-orm. 

```
import Media from '@/models/media'
import media from '@/state/media'

const database = new VuexORM.Database()
database.register(Media, media)
const ormStore = VuexORM.install(database, { namespace: 'database' })
```

Then we could use vuex actions directly from the model import, and use the model import for vuex-orm queries. 
```
import Media from '@/models/media';

Media.dispatch('fetchMedia').then(
  response => {
    if (!response.data.errors) {
      Media.insert({ data: response.data.media })
    }
  }
)
```

Can pinia-orm models and pinia stores be linked so I can access the state/and actions from the model import?

I see there is the piniaStore() method on the model, but even if I give them the same name, they don't seem to be linked. I'm guessing the model goes into it's own name space, and is a seperate store.

I can just import them seperately, it just means that sometimes we have 15 models imports in one file. Now we have another 15 lines of useRepo definitions, and another 15 lines of useStore imports.

```
import Media from '@/models/media';
const MediaRepo = useRepo(Media);
import { useMediaStore } from '@/state/media';

useMediaStore().fetchMedia().then(
  response => {
    if (!response.data.errors) {
      MediaRepo.save(response.data.media)
    }
  }
)
```

Just wanted to clarify that I'm understanding this right. Thanks

**Comment by CodeDredd** [comment, marked as answer]
Hey again 😄 

As you mentioned correctly there is no registration anymore like in `vuex-orm-next`. I was thinking about your case if it makes sense give `useRepo` an option to pass something like `state`, but in my opnionen there is no benefit.

The `piniaStore` method is not on the `Model`. It*s on the `Repository`. `Model` will never have like `vuex-orm-net` business logic because `pinia-orm` and `vuex-orm-next` are follwing the [Data Mapper Pattern](https://en.wikipedia.org/wiki/Data_mapper_pattern)

If it's just about line saving and you use the Repo only once in the code then you can also write:
````js
import { useRepo } from 'pinia-orm'
import Media from '@/models/media';
import { useMediaStore } from '@/state/media';

useMediaStore().fetchMedia().then(
  response => {
    if (!response.data.errors) {
      useRepo(Media).save(response.data.media)
    }
  }
)
````

One line less 😄 .

You could use a helper function if your models are all initlized the same way like:

````js
import { useRepo } from 'pinia-orm'

export function initRepo(model, stateStore, method) {
  stateStore[method]().then(
    response => {
      if (!response.data.errors) {
        useRepo(model).save(response.data.media)
      }
    }
  )
}
````

or you create a [custom repository](https://pinia-orm.codedredd.de/guide/repository/getting-started#custom-repository) where you put this logic in. Which would be semanticly the right choice 😉
